lib/utils/ens/index.ts

```diff
@@ -0,0 +1,16 @@
+import { getDefaultProvider } from "@ethersproject/providers";
+
+export const isENSDomain = (domain: string) =>
+  !!domain && domain.includes(".eth");
+
+export const ETHProvider = getDefaultProvider();
+
+export const getAddressByENS = async (domain: string) => {
+  try {
+    const address = await ETHProvider.resolveName(domain);
+    return address;
+  } catch (e) {
+    console.log("Error in getAddressByENS", e);
+    return Promise.reject(e);
+  }
```

Comment:
```suggestion
    console.error("Error in getAddressByENS", e);
    return Promise.reject(e);
  }
```

Reply:
I usually do `throw e` instead of `return Promise.reject(e)` what's the difference? an outer `await` in another try/catch will still catch right? I guess both achieve the same thing?

Reply:
Yes, both achieve the same thing in the end. But as we are awaiting a promise which could fail and not in a plain try/catch without promises, I decided to go for Promise.reject